src/develop/develop.c

```diff
@@ -1059,15 +1060,36 @@ void dt_dev_add_history_item_ext(
   _dev_add_history_item_ext(dev, module, enable, FALSE, no_image, FALSE);
 }
 
-void _dev_add_history_item(
+static gboolean _dev_undo_start_record_target(dt_develop_t *dev, gpointer target)
+{
+  const double this_time = dt_get_wtime();
+  if(target && target == dev->gui_previous_target
+     && this_time < MIN(dev->gui_previous_time + 15.f,
+                        dev->gui_previous_pipe_time + 5.f))
```

Comment:
When I drag say the RGB curve it feels to me that 5s is really to long to wait for an new undo to be recorded. To me 2s would be better here, that is, if I wait 2s looking at the result of the dragging then I expect to be able to undo at this point.

Tested also while editing masks and 5s is really too long. We cannot undo to some previous point in history even after having waited for say 4s.

Anyway here I recommend to use two (hidden) configuration values. This will document the intent and will make it easier to adjust later if needed.